service/controller/internal/cloudconfig/master_template.go

```diff
@@ -46,19 +48,27 @@ func (c *CloudConfig) NewMasterTemplate(ctx context.Context, data IgnitionTempla
 
 		params = k8scloudconfig.DefaultParams()
 
+		params.BaseDomain = strings.Join([]string{key.ClusterID(data.CustomObject), "k8s", key.BaseDomain(data.CustomObject)}, ".")
 		params.Cluster = data.CustomObject.Spec.Cluster
 		params.DisableEncryptionAtREST = true
 		// Ingress controller service remains in k8scloudconfig and will be
 		// removed in a later migration.
 		params.DisableIngressControllerService = false
-		params.EtcdPort = data.CustomObject.Spec.Cluster.Etcd.Port
+		params.EnableAWSCNI = false
+		params.Etcd = k8scloudconfig.Etcd{
+			ClientPort:          key.EtcdPort(data.CustomObject),
+			HighAvailability:    false,
+			InitialCluster:      "etcd0=https://etcd0." + params.BaseDomain + ":2380",
```

Comment:
I am not sure if this will just work. AFAIK this assumes the structure of HA masters with a single master.

Reply:
True. In k8scloudconfig before HA masters the only etcd node was called `etcd0` so I believe this is backwards-compatible. At least it worked when I tested creation and upgrades. I'm not sure if there are other scenarios where this could fail.